js/serial.js

```diff
@@ -12,10 +15,76 @@ var serial = {
     transmitting:   false,
     outputBuffer:  [],
 
+    LOGHEAD: 'SERIAL: ',
+
     connect: function (path, options, callback) {
         var self = this;
         self.openRequested = true;
 
+        var testUrl = path.match(/tcp:\/\/(.*):(.*)/)
+        if (testUrl) {
+            self.connectionIP = testUrl[1];
+            self.connectionPort = testUrl[2] || self.connectionPort;
+            self.connectionPort = parseInt(self.connectionPort);
+            self.connectionType = 'tcp';
+            self.LOGHEAD = 'SERIAL-TCP: ';
+            console.log('connect to raw tcp:', self.connectionIP + ':' + self.connectionPort)
+
+        chrome.sockets.tcp.create({}, function(createInfo) {
+            console.log('chrome.sockets.tcp.create', createInfo)
+            if (createInfo && !self.openCanceled) {
+                self.connectionId = createInfo.socketId;
+                self.bitrate = 115200; // fake
+                self.bytesReceived = 0;
+                self.bytesSent = 0;
+                self.failed = 0;
+                self.openRequested = false;
+            }
+
+
+            chrome.sockets.tcp.connect(createInfo.socketId, self.connectionIP, self.connectionPort, function (result){
+                if (chrome.runtime.lastError) {
+                    console.error('onConnectedCallback', chrome.runtime.lastError.message);
+                }
+
+                console.log('onConnectedCallback', result)
+                if(result == 0) {
+                    chrome.sockets.tcp.setNoDelay(createInfo.socketId, true, function (noDelayResult){
+                        if (chrome.runtime.lastError) {
+                            console.error('setNoDelay', chrome.runtime.lastError.message);
+                        }
+
+                        console.log('setNoDelay', noDelayResult)
+                        if(noDelayResult != 0) {
+                            self.openRequested = false;
+                            console.log(self.LOGHEAD + 'Failed to setNoDelay');
```

Comment:
No that's all good then. There are cases where apps need to be given special permissions in chrome (i.e. outside of the app), but if this isn't one of them then just logging it to the console is fine.